src/domains/wallet-app/api/npg/v1/_openapi.json.tpl

```diff
@@ -0,0 +1,138 @@
+{
+    "openapi": "3.0.3",
+    "info": {
+      "version": "0.0.1,",
+      "title": "pagoPA Wallet Notification API - OpenAPI 3.0",
+      "description": "Notification API to implement callback providing order status updates after wallet creation on Nexi's NPG",
+      "contact": {
+        "name": "pagoPA - Touchpoints team"
+      }
+    },
+    "tags": [
+      {
+        "name": "Wallet",
+        "description": "Intercat with wallet"
+      }
+    ],
+    "externalDocs": {
+      "description": "Nexi API to handle Hosted Payment Page (HPP) payments",
+      "url": "https://developer.nexigroup.com/it/api/post-orders-hpp"
+    },
+    "servers": [
+      {
+        "url": "https://${hostname}"
+      }
+    ],
+    "paths": {
+      "/notify": {
+        "post": {
+          "tags": [
+            "notifyWallet"
+          ],
+          "summary": "Update Wallet on NPG payment instrument creation",
+          "description": "Update Wallet on NPG payment instrument creation",
+          "operationId": "notifyWallet",
+          "parameters": [
+            {
+              "in": "header",
+              "name": "correlation-Id",
+              "schema": {
+                "type": "string",
+                "format": "uuid"
+              },
+              "required": true,
+              "description": "Unique identifier of the request"
+            }
+          ],
+          "requestBody": {
+            "description": "Notify wallet",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/NotificationRequest"
+                }
+              }
+            },
+            "required": true
+          },
+          "responses": {
+            "204": {
+              "description": "Payment reqeust sent successfully"
+            },
+            "400": {
+              "description": "Invalid input",
+              "content": {
+                "application/json": {
+                  "schema": {
+                    "$ref": "#/components/schemas/ErrorResponse"
+                  }
+                }
+              }
+            },
+            "500": {
+              "description": "Internal server error",
+              "content": {
+                "application/json": {
+                  "schema": {
+                    "$ref": "#/components/schemas/ErrorResponse"
+                  }
+                }
+              },
+            },
+            "405": {
+              "description": "Invalid input"
```

Comment:
This way you only return a string, wouldn’t it be better to return an ErrorResponse object ?

Reply:
Removed this code and used problem json to describe the error